packages/apollo-server-fastify/src/ApolloServer.ts

```diff
@@ -0,0 +1,137 @@
+import * as fastify from 'fastify';
+import { FastifyInstance } from 'fastify';
+const { parseAll } = require('accept');
+import {
+  renderPlaygroundPage,
+  RenderPageOptions as PlaygroundRenderPageOptions,
+} from '@apollographql/graphql-playground-html';
+
+import { graphqlFastify } from './fastifyApollo';
+
+export { GraphQLOptions, GraphQLExtension } from 'apollo-server-core';
+import { ApolloServerBase, GraphQLOptions } from 'apollo-server-core';
+import { IncomingMessage, OutgoingMessage, ServerResponse } from 'http';
+
+export class ApolloServer extends ApolloServerBase {
+  async createGraphQLServerOptions(
+    request: fastify.FastifyRequest<IncomingMessage>,
+    reply: fastify.FastifyReply<OutgoingMessage>,
+  ): Promise<GraphQLOptions> {
+    const options = await super.graphQLServerOptions({ request, reply });
+    return options;
+  }
+
+  protected supportsSubscriptions(): boolean {
+    return true;
+  }
+
+  protected supportsUploads(): boolean {
+    return true;
+  }
+
+  public async applyMiddleware({
+    app,
+    cors,
+    path,
+    route,
+    disableHealthCheck,
+    onHealthCheck,
+  }: ServerRegistration) {
+    if (!path) path = '/graphql';
+
+    await app.addHook(
+      'onRequest',
+      async function(
+        this: any,
+        request: any,
+        response: ServerResponse,
+        next: any,
+      ) {
+        if (request.path !== path) {
+          return next();
+        }
+
+        if (this.playgroundOptions && request.method === 'get') {
+          // perform more expensive content-type check only if necessary
+          const accept = parseAll(request.headers);
```

Comment:
I would use https://github.com/fastify/fastify-accepts instead.

Reply:
Done

Reply:
I don't think this is working right now, the usage is different.

Reply:
@mcollina I need a `next()` method along with `async/await` but based on the docs we can't use `next()` along with `async/await`. What is the alternative in this case?

Reply:
Why do you need it?